tests/TestCase/Controller/ControllerTest.php

```diff
@@ -576,10 +576,6 @@ public function testReferer(): void
         $result = $Controller->referer(['controller' => 'Posts', 'action' => 'index'], true);
         $this->assertSame('/posts/index', $result);
 
-        $request = $this->getMockBuilder(ServerRequest::class)
-            ->onlyMethods(['referer'])
-            ->getMock();
```

Comment:
no idea why this was not detected by static analysis. This variable is overwritten directly bellow.

Reply:
overwriting vars is probably not the main detection topic of those.
especially since they could have side effects inside methods, so hard to fully grasp the superfluousness